tests/setup.go

```diff
@@ -25,6 +25,11 @@ func getLocalConnection() net.Conn {
 	return conn
 }
 
+func getLocalConnectionPtr() *net.Conn {
+	conn := getLocalConnection()
+	return &conn
+}
```

Comment:
We probably don't need this new function, you can likely obtain a reference from the value you get from the `getLocalConnection()` method.

Reply:
Thanks @JyotinderSingh for the review. I was just wondering, if I use getLocalConnection(), wont it be a call by value instead, and copy the net.Conn over to the caller. The reason I was trying to avoid that is because this test is already maintaining a lot of connections, and as the size of the project grows, the tests would get expensive too.

Reply:
We don't run any tests concurrently (since the server state for one particular test may interfere with other unrelated tests). So there isn't much of a concern for having too many connections.
Additionally, even if we pass by pointer we still have almost the same number of resources allocated regarding network IO, memory, etc. So it is not a large difference.

Reply:
Ok, changed it back to getLocalConnection